### ISSUE TITLE
get_all_shortest_paths_dijkstra() should use comparison with tolerances

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1908]
`igraph_get_all_shortest_paths_dijkstra()` should use comparisons with tolerances to produce the same result as betweenness functions. This function returns all shortest paths between pairs of vertices. In the weighted case, it sometimes fails to detect that two paths have the same length due to numerical roundoff errors.

Related:

https://github.com/igraph/igraph/issues/1796

https://github.com/igraph/rigraph/issues/314

**Comment by szhorvat** [comment]
Fixed in b439d1647af4bcf7c0e423c7e981c39fa98ede45

**Comment by szhorvat** [comment]
For reference, this (rather inefficient) Mathematica function find all shortest paths in `g` between `s` and `t` using tolerance `eps`:

```mathematica
allShortestPaths[g_, s_, t_, eps_ : 1.*^-10] :=
 FindPath[g, s, t, GraphDistance[g, s, t]*{1 - eps, 1 + eps}, All]
```

No IGraph/M here, this is pure Mathematica.
